### DISCUSSION TITLE
Using 3D and Light

### DISCUSSION THREAD

**Discussion by acklavidian** [Orama-Interactive/Pixelorama#596] [Ideas]
**Feature/enhancement description:**

I am working on an image editor myself, but I'm wondering if I should just make an extension to an existing editor.
My editor "Rasteri" uses 3D under the hood. My idea is to allow the user to edit pixel art assets with normal maps asynchronously.
![Peek 2020-11-17 13-18](https://user-images.githubusercontent.com/3977726/99689209-b6f41100-2a54-11eb-854d-8edf0c4d80d2.gif)

**Comment by acklavidian** [comment]
I have a lot more features planned, and the gif above doesn't show any depth changes, just light. Essentially under the hood I am using Threejs. Looking for feed back if I should just make Rasteri apart of Pixelorama (...or something else). I already made Godot a plugin while back for generating animated 2D Sprite objects from 3D, and attaching things to them. It's called [Smushgd](https://github.com/acklavidian/smushgd) One advantage godot has over threejs is the quality of CSG Objects.  I CSG would lend themselves to this kind of work

**Comment by acklavidian** [comment]
Here is more progress with rasteri. Still not sure if it will end up as its own editor. 
![update](https://i.imgur.com/TnOVMZJ.gif)

**Comment by zhengying** [comment]
This looks great. it's very helpful for pixel art post effect!
